src/ripple/conditions/impl/utils.h

```diff
@@ -109,34 +127,14 @@ parsePreamble(Slice& s, std::error_code& ec)
     s += 1;
 
     if (p.tag == 0x1F)
-    { // Long tag form:
-        p.tag = 0;
-
-        if (s[0] == 0x80)
-        {
-            ec = error::generic;
-            return p;
-        }
-
-        while (!s.empty() && (s[0] & 0x80))
-        {
-            p.tag = p.tag * 128 + (s[0] & 0x7F);
-            s += 1;
-        }
-
-        if (s.empty())
-        {
-            ec = error::generic;
-            return p;
-        }
-
-        p.tag = p.tag * 128 + s[0];
-        s += 1;
+    { // Long tag form, which we do not support:
+        ec = error::long_tag;
+        return p;
     }
 
     if (s.empty())
```

Comment:
I believe this needs to be moved down either inside the long form ot after it. `s` can never be empty here.